spec/paperclip/attachment_spec.rb

```diff
@@ -31,9 +31,109 @@
     file.close
   end
 
+  context "when generators and procesors are provided" do
+    class TestProcessor < Paperclip::Processor
+      def make
+        original_content = file.tap(&:rewind).read
+        file.rewind
+        result = Tempfile.new('paperclip-test')
+        result.print "#{original_content}#{options[:suffix]}"
+        result.tap(&:rewind)
+      end
+    end
+
+    class TestFullGenerator < Paperclip::Processor
+      def make
+        results = {}
+        original_content = file.tap(&:rewind).read
+        file.rewind
+        options.each do |name, style_options|
+          result = Tempfile.new('paperclip-test')
+          result.print "#{original_content}#{style_options[:suffix].upcase}"
+          results[name] = result.tap(&:rewind)
+        end
+        results
+      end
+    end
+
+    class TestPartGenerator < Paperclip::Processor
+      def make
+        original_content = file.tap(&:rewind).read
+        file.rewind
+        name, style_options = *options.first
+        result = Tempfile.new('paperclip-test')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.